tracing-subscriber/tests/registry_span_instrument_drop.rs

```diff
@@ -0,0 +1,206 @@
+#![cfg(feature = "registry")]
+
+use std::sync::{
+    atomic::{AtomicUsize, Ordering},
+    Arc,
+};
+
+use tracing::{
+    span::{self, Id},
+    Dispatch, Event, Metadata, Subscriber,
+};
+use tracing_core::{Interest, LevelFilter};
+use tracing_subscriber::{
+    layer::{Context, SubscriberExt},
+    Layer, Registry,
+};
+
+#[test]
+fn test_callback_counts() {
+    /// Wraps `tracing_subscriber::Registry` and adds some accounting
+    /// to verify that the subscriber is receiving the expected number of calls.
+    struct CountingSubscriber {
+        inner: Registry,
+        new_count: Arc<AtomicUsize>,
+        clone_count: Arc<AtomicUsize>,
+        enter_count: Arc<AtomicUsize>,
+        exit_count: Arc<AtomicUsize>,
+        close_count: Arc<AtomicUsize>,
```

Comment:
Since the subscriber goes into an `Arc`, we can remove the `Arc` around each atomic.